Source/ObjectiveDropboxOfficial/Shared/Handwritten/Networking/DBGlobalErrorResponseHandler.m

```diff
@@ -135,33 +137,47 @@ + (void)executeRegisteredResponseBlocksWithRouteError:(id)routeError error:(DBRe
   unsigned int count;
   objc_property_t *props = class_copyPropertyList([errorClass class], &count);
 
-  for (int i = 0; i < count; i++) {
+  NSString *tagValue = nil;
+
+  for (unsigned int i = 0; i < count; i++) {
     objc_property_t property = props[i];
     const char *name = property_getName(property);
     NSString *propertyName = [NSString stringWithCString:name encoding:NSUTF8StringEncoding];
+    if ([propertyName isEqualToString:@"tag"]) {
+      tagValue = [routeError tagName];
+      break;
+    }
+  }
 
-    const char *type = property_getAttributes(property);
+  for (unsigned int i = 0; i < count; i++) {
+    objc_property_t property = props[i];
+    const char *name = property_getName(property);
+    NSString *propertyName = [NSString stringWithCString:name encoding:NSUTF8StringEncoding];
 
-    NSString *typeString = [NSString stringWithUTF8String:type];
+    const char *type = property_getAttributes(property);
+    NSString *typeString = [NSString stringWithCString:type encoding:NSUTF8StringEncoding];
     NSArray *attributes = [typeString componentsSeparatedByString:@","];
     NSString *typeAttribute = [attributes objectAtIndex:0];
-    NSString *propertyType = [typeAttribute substringFromIndex:1];
 
     if ([typeAttribute hasPrefix:@"T@"]) {
       NSString *typeClassName =
-          [typeAttribute substringWithRange:NSMakeRange(3, [typeAttribute length] - 4)]; // turns @"NSDate" into NSDate
+          [typeAttribute substringWithRange:NSMakeRange(3, [typeAttribute length] - 4)]; // turns T@"NSDate" into NSDate
       Class typeClass = NSClassFromString(typeClassName);
       if (typeClass != nil && typeClass != [NSString class]) {
-        // we try / catch here because for union types, a runtime exception is thrown
-        // when a value type that doesn't correspond to the tag state is accessed.
         @try {
+          // we want to make sure that we only access fields that correspond to the
+          // correct Union tag state. This check should filter most cases, but because
+          // of the imprecision of reflection, we still want the try catch block
+          if (tagValue && ![tagValue localizedCaseInsensitiveContainsString:propertyName]) {
```

Comment:
Can you add a comment about why this is a contains rather than equals? That's not clear to me from reading the code. Also, I think you should use a simpler comparison rather than a localized one, since the tags are probably not localized, right?